integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/IntegrasjonspunktImpl.java

```diff
@@ -85,31 +83,35 @@ public GetCanReceiveMessageResponseType getCanReceiveMessage(@WebParam(name = "G
             return response;
         }
 
-        boolean certificateAvailable = adresseRegister.hasAdresseregisterCertificate(serviceRecord);
-
         final LogstashMarker marker = MarkerFactory.receiverMarker(organisasjonsnummer);
+        boolean validServiceIdentifier = false;
         boolean mshCanReceive = false;
         boolean isDpv = false;
-        if (certificateAvailable) {
+        if (asList(DPO, DPI, DPF).contains(serviceRecord.getServiceIdentifier()) &&
+                strategyFactory.hasFactory(serviceRecord.getServiceIdentifier())) {
+            validServiceIdentifier = true;
             Audit.info("CanReceive = true", marker);
         } else if (mshClient.canRecieveMessage(organisasjonsnummer)) {
             mshCanReceive = true;
             Audit.info("MSH canReceive = true", marker);
-        } else if (asList(DPV, DPE_INNSYN).contains(serviceRecord.getServiceIdentifier())) {
+        } else {
             isDpv = true;
         }
 
-        if (!mshCanReceive && !certificateAvailable && !isDpv) {
-            Audit.error("CanReceive = false", marker);
+        boolean strategyFactoryAvailable;
+        if (isDpv) {
+            strategyFactoryAvailable = strategyFactory.hasFactory(DPV);
+        } else {
+            strategyFactoryAvailable = validServiceIdentifier;
         }
-
-        boolean strategyFactoryAvailable = strategyFactory.hasFactory(serviceRecord.getServiceIdentifier());
         if (!strategyFactoryAvailable && !mshCanReceive) {
-            Audit.error(String.format("StrategyFactory for %s not found. Feature toggle might be disabled.",
-                    serviceRecord.getServiceIdentifier()), marker);
+            Audit.error(String.format("CanReceive = false. Either feature toggle for %s is disabled, or MSH cannot receive",
+                    isDpv ? DPV : serviceRecord.getServiceIdentifier()), marker);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Change this condition so that it does not always evaluate to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS2583)